### PR TITLE
[INFINITY-1810] Add integration tests for verifying metrics

### PR DESCRIPTION
**Pull request by NimaVaziri**
Introduces a new utility package for testing functionality of metrics.

### CODE REVIEW

**Comment by NimaVaziri** [comment]
retest this please

**Comment by mellenburg** [comment]
@NimaVaziri Hey, sorry, this PR has been littered with fault status's Do not try rerunning the ones linked here.

File: testing/sdk_metrics.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+'''Utilities relating to checking services' metrics'''
```
Comment (jsrodman) [inline testing/sdk_metrics.py]: I'd like this to say something about what's being tested here.
eg "verifies metrics functionality by fetching metrics from the DC/OS metric component"

File: testing/sdk_metrics.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    Keyword arguments:
+    service_name -- the name of the service to get metrics for
+    task_name -- the name of the task whose agent to run metrics commands from
+    """
+    host = hosts.system_host(service_name, task_name)
+    auth_token = shakedown.run_dcos_command('config show core.dcos_acs_token')[0].strip()
+
+    service_containers_cmd = """curl --header "Authorization: token={}"
+        -s http://localhost:61001/system/v1/metrics/v0/containers""".format(auth_token).replace("\n", "")
```
Comment (jsrodman) [inline testing/sdk_metrics.py]: I'm surprised there's no shakedown or dcos commandline utility to help with fetching these urls.

File: testing/sdk_metrics.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    output = shakedown.run_command_on_agent(host, service_containers_cmd)[1]
```
Comment (jsrodman) [inline testing/sdk_metrics.py]: Don't care about failure?

Style nitpick; if you really don't care about failure,
``` 
_, output = shakedown.run_command... 
```
is clearer than [1]

File: testing/sdk_utils.py [hunk 4]
```diff
@@ -39,9 +40,12 @@ def try_throws_n_times(predicate, attempts=1, sleep_seconds=0):
     for index in range(attempts):
         try:
             return predicate()
+        except dcos.errors.DCOSException as e:
+            if str(e) == "Package is already installed":
+                return
```
Comment (jsrodman) [inline testing/sdk_utils.py]: Feels like the wrong solution to the problem.  What if it's the first try?

**Comment by NimaVaziri** [comment]
@jsrodman addressed comments

File: frameworks/kafka/tests/test_sanity.py [hunk 5]
```diff
@@ -273,6 +276,16 @@ def test_pods_cli():
     assert service_cli('pods status {}-0'.format(DEFAULT_POD_TYPE), service_name=FOLDERED_SERVICE_NAME)
     assert service_cli('pods info {}-0'.format(DEFAULT_POD_TYPE), service_name=FOLDERED_SERVICE_NAME, print_output=False) # noisy output
 
+@pytest.mark.sanity
+@pytest.mark.metrics
+def test_metrics():
+    def metrics_exist():
```
Comment (nickbp) [inline frameworks/kafka/tests/test_sanity.py]: Could just have a `wait_for_any_metrics(service_name, node_name, timeout=600)` function in `sdk_metrics` (containing the current content of these tests), and then have these tests just call `wait_for_any_metrics`.

**Pull request merged**